torchtitan/parallelisms/parallelize_llama.py

```diff
@@ -413,13 +413,27 @@ def apply_tp(
             parallelize_plan=layer_plan,
         )
 
+    # updates expressly for async tensor parallel
     if job_config.experimental.enable_async_tensor_parallel:
         from torch.distributed._symmetric_memory import enable_symm_mem_for_group
 
+        torch._dynamo.config.cache_size_limit = 10000
```

Comment:
I wonder if we should unconditionally set this since we are doing per-transformer block compile. @Chillee do you know if it's the right thing to do?

Reply:
good point and happy to make this more refined as atm no guarantee the setting above is optimal, but something like it is required to avoid the silent fail.  
I'm going to go ahead and merge as -s since it's required atm to avoid users from outright silent failing when they enable async_tp, and happy to update with more refined limit(s) based on @Chillee feedback.

Reply:
IMO the right thing to do here is to avoid recompilation for per-TransformerBlock compile, which requires fixing this https://github.com/pytorch/pytorch/issues/125836 and enable inline_built_nn_modules by default